primitives/bridge/src/symbiotic_message_processor.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use frame_support::pallet_prelude::*;
+use parity_scale_codec::DecodeAll;
+use snowbridge_core::Channel;
+use snowbridge_router_primitives::inbound::envelope::Envelope;
+use snowbridge_router_primitives::inbound::MessageProcessor;
+use sp_runtime::DispatchError;
+use sp_std::vec::Vec;
+
+/// Magic bytes are added in every payload intended for this processor to make sure
+/// that we are the intended recipient of the message. Reason being scale encoding is not type aware.
+/// So a same set of bytes can be decoded for two different data structures if their
+/// total size is same. Magic bytes can be checked after decoding to make sure that the sender
+/// indeed send a message intended for this processor.
+pub const MAGIC_BYTES: [u8; 4] = [112, 21, 0, 56];
+
+#[derive(Encode, Decode)]
+pub struct Payload<T>
+where
+    T: pallet_external_validators::Config,
+{
+    pub magic_bytes: [u8; 4],
+    pub message: Message<T>,
+}
+
+#[derive(Encode, Decode)]
+pub enum Message<T>
+where
+    T: pallet_external_validators::Config,
+{
+    V1(Command<T>),
+}
+
+#[derive(Encode, Decode)]
+pub enum Command<T>
```

Comment:
shal we call this InboundCommand? reason is we have a different est of outbound commands right?

Reply:
*fixed*